Optimization Attempt 1

djason-todd opened the pull request:
Prospective optimizations to reduce latency

djason-todd committed:
Update nmea_logger.py

Attempted to add a buffer to decrease latency, as well as a few other improvements. See 'Code Optimization Ideas for elaboration.

Signed-off-by: djason-todd <112493630+djason-todd@users.noreply.github.com>

djason-todd committed:
Add files via upload

Signed-off-by: djason-todd <112493630+djason-todd@users.noreply.github.com>

djason-todd committed:
Update setSysClockFromGGA.py

Signed-off-by: djason-todd <112493630+djason-todd@users.noreply.github.com>

djason-todd committed:
Merge pull request #1 from djason-todd/djason-todd-patch-1

Update nmea_logger.py

djason-todd committed:
Merge pull request #2 from djason-todd/djason-todd-patch-2

Update setSysClockFromGGA.py

SBFRF commented on DataAcq/setSysClockFromGGA.py:
looks great

SBFRF commented on DataAcq/nmea_logger.py:
```diff
@@ -2,39 +2,42 @@
+    
+    if not os.path.exists(path):
+        os.makedirs(path)
+        print(f"Made Directory {path}")
```
we should migrate to the `logging` package to provide info to user.  e.g. `logging.info('Made directory path') ` or `logging.warning(___)`

SBFRF replied:
@djason-todd  did this get resolved for all print statements?

SBFRF commented on DataAcq/nmea_logger.py:
```diff
@@ -2,39 +2,42 @@
-        file1.write('###' + datestr)
+    buffer = []
+
+    while True:
```
:)

SBFRF commented on DataAcq/nmea_logger.py:
```diff
@@ -2,39 +2,42 @@
+
+    while True:
+        line = ser.readline().decode('ascii', errors='replace')  # first read the line
+        datestr = str(datetime.datetime.now())
```
is there a faster way to get "now" than `datetime.now` ?  i'm thinking about epoch time ... it's usually faster than creating a `datetime` object. .... this would require a re-write of our data reading scripts

SBFRF replied:
is it necessary?

djason-todd replied:
Which files would require a rewrite if we went about integrating epoch time? Knowing that would help me figure out the next best course of action.

--Do not spend a lot of time if the current improvements are not too burdensome.
--files are in a different repo/diff location in the same repo; TBD

djason-todd replied:
And how do you mean is it necessary? Are you thinking of going about removing the process entirely or am I misunderstanding?

--referring to the transition between datenow.now to epoch time, not the removal of any singular processes within the code.

djason-todd replied:
Decided to table due to downstream processing, as mentioned in discussion.

SBFRF reviewed (commented):
ok try this!

SBFRF commented:
I think we talked about is it necessary (is it worth going through all the effort)

The maintained read files are in another repo

djason-todd committed:
Update nmea_logger.py

Migrated to the logging package, kept the datetime.now to be compared against epoch time at a later date.

Signed-off-by: djason-todd <112493630+djason-todd@users.noreply.github.com>

djason-todd committed:
Merge pull request #3 from djason-todd/djason-todd-nmealoggingpackage_migration

Update nmea_logger.py

SBFRF commented:
@djason-todd  i'm seeing that there are conflicts that need to be resolved external to web environment

djason-todd committed:
Update setSysClockFromGGA.py

Signed-off-by: djason-todd <112493630+djason-todd@users.noreply.github.com>

djason-todd committed:
Update setSysClockFromGGA.py

Signed-off-by: djason-todd <112493630+djason-todd@users.noreply.github.com>

SBFRF reviewed (commented):
looks great! thanks for doing this, standing by for report on stability before approving/merging
